app/models/course.rb

```diff
@@ -53,6 +52,8 @@ class Course < ActiveRecord::Base
                                     dependent: :destroy
   has_many :lesson_plan_events, through: :lesson_plan_items,
                                 source: :actable, source_type: Course::LessonPlan::Event.name
+  # Achievements must be declared after material_folders or duplication will fail.
+  has_many :achievements, dependent: :destroy
```

Comment:
Why would it fail if `:achievements` is declared above `material_folders`? 

Also, the service spec tests components in isolation - it would be good to include one which guards against the ordering of this.